### ISSUE TITLE
`Lint/AssignmentInCondition` gets fooled by parentheses

### ISSUE THREAD

**Issue by Drenmi** [bbatsov/rubocop#4842]
This cop can be tricked by simply adding parentheses around the condition. E.g.:

```ruby
foo if (bar = baz)
```

--------

## Expected behavior

RuboCop registers an offense.

## Actual behavior

RuboCop doesn't register an offense.

## Steps to reproduce the problem

Run `bundle exec rubocop` on:

```ruby
foo if (bar = baz)
```

## RuboCop version

0.50.0 (using Parser 2.4.0.0, running on ruby 2.4.2 x86_64-darwin16)

**Comment by Drenmi** [comment]
Arguably the author could use parentheses to indicate they know what they're doing, so there might be potential for an `AllowParenthesized` configuration option.

**Comment by Drenmi** [comment]
Never mind. There's an `AllowSafeAssignment` option that does just that.
